Extensions/xkit_patches.js

```diff
@@ -915,6 +915,209 @@ XKit.extensions.xkit_patches = new Object({
+
+				control_button_template: null,
+				get_control_button_template: async function() {
+					await XKit.css_map.getCssMap();
+
+					var selector = XKit.css_map.keyToClasses("controlIcon").map(css => `[data-id]:first footer .${css}:first`).join(", ");
+					var control = $(selector);
+
+					var get_used_class_from_map = function(key) {
+						const keyCss = XKit.css_map.keyToCss(key);
+						var element = control.find(keyCss);
+
+						return element.attr("class");
+					};
+
+					var controlIconClass = control.attr("class");
+					var buttonClass = get_used_class_from_map("button");
+
+					var new_control = `
+						<div class="${controlIconClass} {{className}} xkit-interface-control-button" title="{{text}}" {{additional}}>
+							<button class="${buttonClass}" aria-label="" tabindex="0">
+								<div class="xkit-interface-icon" {{data}}></div>
+							</button>
+						</div>
+					`;
+
+					this.control_button_template = new_control;
+
+					return this.control_button_template;
+				},
+				/**
+				 * Create a specification for a control button that can be added to
+				 * future posts using `XKit.interface.add_control_button`.
+				 * @param {String} class_name - CSS class of the button to be created
+				 * @param {String} icon - URL of the button's icon
+				 * @param {String} text - Hover text of the button
+				 * @param {EventListener} func - Function called on click of control button
+				 * @param {String?} ok_icon - URL of icon displayed when the button is
+				 *                            "completed" (e.g. reblog button turning green)
+				 */
+				create_control_button: async function(class_name, icon, text, func, ok_icon) {
+					if (this.control_button_template == null) {
+						this.control_button_template = await this.get_control_button_template();
+					}
+
+					XKit.interface.added_icon.push(class_name);
+					XKit.interface.added_icon_icon.push(icon);
+					XKit.interface.added_icon_text.push(text);
+
+					XKit.tools.add_css(`.${class_name} .xkit-interface-icon {
+						background-image: url('${icon}');
+						background-size: 100% 100%;
+						width: 21px;
+						height: 21px;
+					}`, `xkit_interface_icon__${class_name}`);
+
+					if (typeof ok_icon !== "undefined") {
+						XKit.tools.add_css(`.${class_name} .xkit-interface-icon-completed {
+							background-image: url('${ok_icon}');
+						}`, `xkit_interface_icon__completed__${class_name}`);
+					}
+
+					$(document).on('click', '.' + class_name, function() {
+						if ($(this).hasClass("xkit-interface-working") || $(this).hasClass("xkit-interface-disabled")) { return; }
+						if (typeof func === "function") { func.call(this, event); }
+					});
+				},
+				add_control_button: async function(obj, class_name, additional) {
+					if (typeof additional == "undefined") {additional = ""; }
+
+					if (XKit.interface.added_icon.indexOf(class_name) === -1) {
+						// console.log("Interface -> Can't add icon, button not created, use create_control_button.");
```

Comment:
```suggestion
```

Reply:
lol, I debated about taking this out, but since I was copying the function over I didn't change what I didn't need to. ah well!